llvm/test/TableGen/aie-memory-cycles.td

```diff
@@ -75,3 +75,13 @@ let Itinerary = II_OTHER in
 // CHECK-NEXT:  return 5;
 // CHECK-LABEL: int TestAIEInstrInfo::getMaxLastMemoryCycle() const {
 // CHECK-NEXT:  return 11;
+
+// CHECK:      std::optional<std::vector<int>>
```

Comment:
That's a rather expensive type to return. How about a `SmallVector<int, 2>`? I don't think we need the optional, for instructions that do not access memory, we can just return an empty vector.